functions/setup_directories.R

```diff
@@ -0,0 +1,408 @@
+# Update - August 2024
+#
+# Written/run on - RStudio Server
+# Version of R - 4.1.2
+#
+# Description - Functions to set up working directory and declare file paths for
+#               use in MI and Publication reports.
+################################################################################
+
+## Generic setup ##-------------------------------------------------------------
+
+# Use general file path functions
+source(here::here("functions/setup_general.R"))
+
+# Use write file function for writing files to disk and setting correct permissions
+source(here::here("functions/write_file.R"))
+
+### MI directory setup ###------------------------------------------------------
+
+#' Set up Management Information Directory
+#' 
+#' @description the function will use the fs package to return the file path to
+#' the management-report directory. 
+#'
+#' @return the path to the management report folder.
+#' 
+#'
+get_mi_dir <- function() {
+  mi_dir <- fs::path("/", "conf", "dementia", "A&I", "Outputs", "management-report")
+  
+  return(mi_dir)
+}
+
+
+#' Set up the Management Information year directory
+#'
+#' @description The function will use the fs package to return the file path to the
+#' management report year directory
+#'
+#' @param folder supply the data or output folder 
+#'
+#' @return the path to the management report year specific folder. 
+#' 
+#'
+get_mi_year_dir <- function(folder = c("data", "output"), test_output = FALSE) {
+  year <- stringr::str_glue("{fy}-{substr(as.numeric(fy)+1, 3, 4)}")
+  qtr <- stringr::str_glue("Q{qt}")
+  test <- "test"
+  
+  if ((test_output)){
+  year_dir <- fs::path(get_mi_dir(), {{ folder }}, year, qtr, test)
+  }else{
+  year_dir <- fs::path(get_mi_dir(), {{ folder }}, year, qtr)
+  }
+  
+  if (!fs::dir_exists(year_dir)) {
+    fs::dir_create(year_dir)
+    cli::cli_alert_info(
+      "{.path {year_dir}} did not exist, it has now been created."
+    )
+  }
+  
+  return(year_dir)
+}
+
+
+#' Get the path to the MI data outputs
+#'
+#' @description The function will return the file path to the data files needed
+#' to create the MI report. 
+#'
+#' @param type Supply the type of data output
+#' @param ext Supply the file extension
+#'
+#' @return the file path to the data files needed to create the MI report.
+#' 
+get_mi_data_path <- function(type = c(
+  "error_data",
+  "dupe_data",
+  "clean_data",
+  "ldp_data",
+  "final_data"
+),
+ext = c("rds", "csv"), 
+test_output = FALSE) {
+  file_name <- dplyr::case_match(
+    type,
+    "error_data" ~ stringr::str_glue("{fy}-{qt}_error-summary"),
+    "dupe_data" ~ stringr::str_glue("{fy}-{qt}_dupes"),
+    "clean_data" ~ stringr::str_glue("{fy}-{qt}_clean-data"),
+    "ldp_data" ~ stringr::str_glue("{fy}-{qt}_individuals-with-ldp"),
+    "final_data" ~ stringr::str_glue("{fy}-{qt}_final-data")
+  )
+  
+  mi_data_path <- get_file_path(
+    directory = get_mi_year_dir("data", test_output = test_output),
+    file_name = file_name,
```

Comment:
I believe we need to add the following after line 101:
`check_mode = "write"`
so that new data files can be created.